src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs

```diff
@@ -46,22 +72,37 @@ public class CsvStreamSource : ISmugglerSource, IDisposable
 
         private readonly List<IDisposable> _disposables = new List<IDisposable>();
 
-        public CsvStreamSource(DocumentDatabase database, Stream stream, DocumentsOperationContext context, string collection)
+        public CsvStreamSource(DocumentDatabase database, Stream stream, DocumentsOperationContext context, string collection, CsvImportOptions csvConfig)
         {
             _database = database;
             _stream = stream;
             _context = context;
             _currentType = DatabaseItemType.Documents;
             _collection = collection;
+            
+            _csvHelperConfig = new CsvConfiguration(CultureInfo.InvariantCulture);
+            
+            _csvHelperConfig.Delimiter = csvConfig.Delimiter;
+            _csvHelperConfig.Quote = csvConfig.Quote;
+            _csvHelperConfig.TrimOptions = csvConfig.TrimOptions;
+            _csvHelperConfig.AllowComments = csvConfig.AllowComments;
+            if (csvConfig.AllowComments && csvConfig.Comment != default(char))
+            {
+                _csvHelperConfig.Comment = csvConfig.Comment;
+            }
+            _csvHelperConfig.BadDataFound = null;
```

Comment:
Why are we setting this to null? It is 'default'?

Reply:
* You can set field `BadDataFound` with a method that is called when bad field data is found.
   You set it to `null `if you don't want an exception to be thrown in case of 'bad data'.

* From the CSVHelper docs:
```    /// <summary>
    /// Gets or sets the function that is called when bad field data is found. A field
    /// has bad data if it contains a quote and the field is not quoted (escaped).
    /// You can supply your own function to do other things like logging the issue
    /// instead of throwing an exception.
    /// Arguments: context
    /// </summary>
    public virtual Action<ReadingContext> BadDataFound { get; set; } = new Action<ReadingContext>(ConfigurationFunctions.BadDataFound);
```

And:
```    public static void BadDataFound(ReadingContext context)
    {
      throw new BadDataException(context, "You can ignore bad data by setting BadDataFound to null.");
    }
```